src/healthcheck/HealthChecker.ts

```diff
@@ -286,7 +286,11 @@ class Plugin {
         })
         .catch((error) => {
           this.status = failure;
-          this.statusReason = error.message;
+          if ((typeof error) === "string") {
+            this.statusReason = error;
+          } else if(error !== undefined) {
```

Comment:
Maybe consider flipping the logic in this if statement. There is no guarantee that if `error !== defined` that `error.message` is present/accessible

Reply:
Would this alternative be preferable?

```js
if (error instanceof Error) {
    this.statusReason = error.message;
} else if ((typeof error) === "string") {
    this.statusReason = error;
}
```

Reply:
Perhaps the following? Then whatever the user passes in the "reject" gets passed through? 
```js
if (error instanceof Error) {
    this.statusReason = error.message;
} else if (error !== undefined) {
    this.statusReason = error;
}
```